### ISSUE TITLE
app crashing

### ISSUE THREAD

**Issue by cylian914** [macarooni-man/auto-mcs#23]
Hey, the app is crashing, here the log (was run with gdb, crash at the same place without):

```
Starting program: /home/cylian91/Prog/trash/auto-mcs 
[Thread debugging using libthread_db enabled]                                                                                                                                                                     
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[186501] PyInstaller Bootloader 5.x
[186501] LOADER: executable is /home/cylian91/Prog/trash/auto-mcs
[186501] LOADER: homepath is /home/cylian91/Prog/trash
[186501] LOADER: _MEIPASS2 is NULL
[186501] LOADER: archivename is /home/cylian91/Prog/trash/auto-mcs
[186501] LOADER: Cookie found at offset 0x2DB2303
[186501] SPLASH: Found splash screen resources.
[186501] SPLASH: Load Tcl library from: /tmp/_MEIrTlDmI/__splash/libtcl8.6.so
[186501] SPLASH: Load Tk library from: /tmp/_MEIrTlDmI/__splash/libtk8.6.so
[186501] LOADER: Loaded functions from tcl/tk libraries.                                                                                                                                                          
[New Thread 0x7ffff69ff6c0 (LWP 186502)]
[186501] SPLASH: Created thread for tcl interpreter.
[186501] SPLASH: Running tcl version 8.6.12 and tk version 8.6.12.
[186501] SPLASH: Splash screen started.                                                                                                                                                                           
[186501] LOADER: Extracting binaries
[New Thread 0x7ffff61fe6c0 (LWP 186506)]
[186501] LOADER: Executing self as child
[186501] LOADER: set _MEIPASS2 to /tmp/_MEIrTlDmI
[186501] LOADER: linux: storing process name into _PYI_PROCNAME: auto-mcs
[186501] LOADER: LD_LIBRARY_PATH=/tmp/_MEIrTlDmI
[Detaching after fork from child process 186507]
[186501] LOADER: Registering signal handlers
[186507] PyInstaller Bootloader 5.x
[186507] LOADER: executable is /home/cylian91/Prog/trash/auto-mcs
[186507] LOADER: homepath is /home/cylian91/Prog/trash
[186507] LOADER: _MEIPASS2 is /tmp/_MEIrTlDmI
[186507] LOADER: archivename is /home/cylian91/Prog/trash/auto-mcs
[186507] LOADER: Cookie found at offset 0x2DB2303
[186507] LOADER: restoring linux process name from _PYI_PROCNAME: auto-mcs
[186507] LOADER: Already in the child - running user's code.
[186507] LOADER: Python library: /tmp/_MEIrTlDmI/libpython3.9.so.1.0
[186507] LOADER: Loaded functions from Python library.
[186507] LOADER: Manipulating environment (sys.path, sys.prefix)
[186507] LOADER: sys.prefix is /tmp/_MEIrTlDmI
[186507] LOADER: Pre-init sys.path is /tmp/_MEIrTlDmI/base_library.zip:/tmp/_MEIrTlDmI/lib-dynload:/tmp/_MEIrTlDmI
[186507] LOADER: Setting runtime options
[186507] LOADER: Initializing python
[186507] LOADER: Overriding Python's sys.path
[186507] LOADER: Post-init sys.path is /tmp/_MEIrTlDmI/base_library.zip:/tmp/_MEIrTlDmI/lib-dynload:/tmp/_MEIrTlDmI
[186507] LOADER: Setting sys.argv
[186507] LOADER: setting sys._MEIPASS
[186507] LOADER: importing modules from CArchive
[186507] LOADER: extracted struct
[186507] LOADER: running unmarshalled code object for struct...
[186507] LOADER: extracted pyimod01_archive
[186507] LOADER: running unmarshalled code object for pyimod01_archive...
[186507] LOADER: extracted pyimod02_importers
[186507] LOADER: running unmarshalled code object for pyimod02_importers...
[186507] LOADER: extracted pyimod03_ctypes
[186507] LOADER: running unmarshalled code object for pyimod03_ctypes...
[186507] LOADER: Installing PYZ archive with Python modules.
[186507] LOADER: PYZ archive: PYZ-00.pyz
[186507] LOADER: Running pyiboot01_bootstrap.py
[186507] LOADER: Running pyi_rth_inspect.py
[186507] LOADER: Running pyi_rth_subprocess.py
[186507] LOADER: Running pyi_rth_pkgres.py
[186507] LOADER: Running pyi_rth_pkgutil.py
[186507] LOADER: Running pyi_rth__tkinter.py
[186507] LOADER: Running pyi_rth_kivy.py
[186507] LOADER: Running pyi_rth_gstreamer.py
[186507] LOADER: Running pyi_rth_multiprocessing.py
[186507] LOADER: Running wrapper.py
[186501] LOADER: Restoring signal handlers
[186501] LOADER: freeing args
[186501] LOADER: child received signal 11; storing for re-raise after cleanup...
[186501] LOADER: Back to parent (RC: 1)
[186501] LOADER: Doing cleanup
[Thread 0x7ffff69ff6c0 (LWP 186502) exited]
[Thread 0x7ffff61fe6c0 (LWP 186506) exited]
[186501] LOADER: Freeing archive status for /home/cylian91/Prog/trash/auto-mcs
[186501] LOADER: re-raising child signal 11
```

i use arch btw

also while you should rly consider getting away of TK, its very unstable, slow and not  great at all for big project. You should consider using  qt (lib called pyside6).

**Comment by macarooni-man** [comment]
Hello there, thanks for bringing the crash to my attention. Can you post the crash report? It should be in `~/.auto-mcs/Logs/`

If a report does not exist, can you describe what you were doing when it crashed?

**Comment by cylian914** [comment]
The report doesn't exist and i just try to start the app.  Also logs should be local (in ./) not in `~/`

**Comment by macarooni-man** [comment]
@cylian914 Thanks for the info! Can you tell me what distribution you are using (stock arch, Manjaro, Garuda, etc.) and the desktop environment?

**Comment by cylian914** [comment]
arch, hyprland (wayland), 6.6.8-arch1-1

**Comment by macarooni-man** [comment]
@cylian914 If you run auto-mcs with the debug flag `./auto-mcs --debug` it should dump a file called `error.log`. If you try this, does the file provide any information? If so, would you be willing to attach it?

**Comment by cylian914** [comment]
still no new file

**Comment by macarooni-man** [comment]
Fixed with #26, testing on more distributions, though this issue is resolved now.
